lib/block.js

```diff
@@ -7,35 +7,39 @@
  */
 
 const TwoPowTwenty = 1048575
+
 /**
  *
- * @param {Number} sequenceNumber The block sequence number.
- * @param {Boolean} moreBlocks If there are more blocks to follow
- * @param {Number} blockSize
+ * @param numOrBlockState The block sequence number or a block state object.
+ * @param more Can indicate if more blocks are to follow.
+ * @param size The block size.
  */
-module.exports.generateBlockOption = (sequenceNumber, moreBlocks, blockSize) => {
-    if (typeof sequenceNumber === 'object') {
-        moreBlocks = sequenceNumber.moreBlocks
-        blockSize = sequenceNumber.blockSize
-        sequenceNumber = sequenceNumber.sequenceNumber
+module.exports.generateBlockOption = (numOrBlockState, more, size) => {
+    let num
+    if (typeof numOrBlockState === 'object') {
+        num = numOrBlockState.num
+        more = numOrBlockState.more
+        size = numOrBlockState.size
+    } else {
+        num = numOrBlockState
     }
 
-    if (sequenceNumber == null || blockSize == null || sequenceNumber == null) {
+    if (num == null || more == null || size == null) {
```

Comment:
if you want to make sure these are numbers maybe check that instead of == null

Reply:
Oh, good point!